src/librustc_driver/driver.rs

```diff
@@ -934,6 +934,7 @@ pub fn phase_3_run_analysis_passes<'tcx, F, R>(sess: &'tcx Session,
     passes.push_pass(MIR_CONST, mir::transform::rustc_peek::SanityCheck);
 
     // What we need to run borrowck etc.
+    passes.push_pass(MIR_CONST, mir::transform::borrow_check::BorrowckMir);
```

Comment:
Well, one reason is that borrowck produces *output* that other things consume. If nothing else, it produces data about which `mut` declarations are used that is later gathered up by the lints.

With respect to typeck, I expect it to be eventually subsumed, most likely, into the region inference code, which will be a part of borrowck, and hence to go away as a distinct entity.